tiny_dnn/activations/softmax_layer.h

```diff
@@ -0,0 +1,58 @@
+/*
+    Copyright (c) 2017, Taiga Nomi
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+#include "tiny_dnn/activations/activation_layer.h"
+#include "tiny_dnn/layers/layer.h"
+
+namespace tiny_dnn {
+
+class softmax_layer : public activation_layer {
+ public:
+  using activation_layer::activation_layer;
+
+  std::string layer_type() const override { return "softmax-activation"; }
+
+  void forward_activation(const vec_t &x, vec_t &y) override {
+    float_t alpha = *std::max_element(x.begin(), x.end());
```

Comment:
> Too C using const char* no?
if you really want to hide a naked pointer, adding `string_ref` class can be a solution.
But I think `const char*` is simple enough in this case.